jupyter-org-client.el

```diff
@@ -991,9 +1021,18 @@ parsed, wrap DATA in a minipage environment and return it."
       (jupyter-org--parse-latex-element data)
     (jupyter-org-export-block "latex" data)))
 
-(cl-defmethod jupyter-org-result ((_mime (eql :text/html)) _params data
+(cl-defmethod jupyter-org-result ((_mime (eql :text/html)) params data
                                   &optional _metadata)
-  (jupyter-org-export-block "html" data))
+  (if (member "pandoc" (alist-get :result-params params))
+      (jupyter-org-scalar
```

Comment:
On further testing, `jupyter-org-scalar` seems to append an extra line if `data` is a short string (but not if it's a long string wrapped in a quote block). It seems to be necessary when data is a table though. I'm slightly confused when I should be calling `jupyter-org-scalar`...

Reply:
Yes `jupyter-org-scalar` should only be passed an unformatted string or table. It is not intended to handle `org` formatted strings. If you want to handle tables specially you should wrap the `org-tabel-to-lisp` call with `jupyter-org-scalar` and the `buffer-string` call with `jupyter-org-raw-string`.

Couldn't you avoid doing all of the table processing by just doing something like

```emacs-lisp
(jupyter-org-raw-string (jupyter-pandoc-convert "html" "org" data))
```

Reply:
Reason I was using table parsing and `jupyter-org-scalar`, is that `jupyter-org-raw-string` inserts the table in a `:RESULTS:` drawer, and I was trying to replicate the behavior of `jupyter-python` and `ob-python` which doesn't have a drawer. However, I think you're right, it's better to be simple, so I'll switch to `jupyter-org-raw-string` without the table parsing.

Longer term, it might be nice to have finer-grained control over whether the table should go in a drawer. For example, I prefer a drawer for big tables but not small ones. But this is not a priority for me, and should probably be a separate issue.

Reply:
I think you can get around wrapping the table with a drawer by also putting an `org-table` text property on the 1st character of the table string. The result won't be wrapped in a drawer if it can be removed by `org-babel` and determining if a result needs to be wrapped is mainly handled by the function `jupyter-org-babel-result-p`.

It would probably be good to create a new function to add this text property similar to what `jupyter-org-raw-string` and `jupyter-org-raw-string-p` do but for the `org-table` property. When doing this we will have to replace all instances of the `org-table` property with the corresponding functions.